packages/components/src/site-thumbnail/test/site-thumbnail.test.jsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { shallow } from 'enzyme';
+import { SiteThumbnail, SITE_THUMBNAIL_DIMENSIONS } from '..';
+
+const CLASS_NAME = 'site-thumbnail__image';
+const IMG_SELECTOR = `.${ CLASS_NAME }`;
+const MSHOTS_URL = 'https://fakeUrl';
+
+describe( 'SiteThumbnail', () => {
+	test( 'has an image that points to mshot', () => {
+		const siteThumbnail = shallow( <SiteThumbnail mShotsUrl={ MSHOTS_URL } /> );
+		const img = siteThumbnail.find( IMG_SELECTOR );
+		expect( img.exists() ).toEqual( true );
+
+		expect(
+			img
+				.prop( 'src' )
+				.includes( 'https://s0.wp.com/mshots/v1/' + encodeURIComponent( MSHOTS_URL ) )
+		).toEqual( true );
+	} );
+
+	test( 'should show the dimension width for the default sizes', () => {
+		const dimension = {
+			width: 100,
+			height: 100,
+		};
+		const siteThumbnail = shallow(
+			<SiteThumbnail mShotsUrl={ MSHOTS_URL } dimension={ dimension } />
+		);
+		const img = siteThumbnail.find( IMG_SELECTOR );
+		expect( img.prop( 'sizes' ) ).toEqual( `${ dimension.width }px` );
+	} );
+
+	test( 'should use the sizesAttr prop as sizes attr', () => {
+		const sizesAttr = '(max-width: 400px) 100vw, 400px';
+		const siteThumbnail = shallow(
+			<SiteThumbnail mShotsUrl={ MSHOTS_URL } sizesAttr={ sizesAttr } />
+		);
+		const img = siteThumbnail.find( IMG_SELECTOR );
+		expect( img.prop( 'sizes' ) ).toEqual( sizesAttr );
+	} );
+
+	test( 'should generate responsive size alternatives 2x and 3x srcset', () => {
+		const dimension = SITE_THUMBNAIL_DIMENSIONS.medium;
+		const siteThumbnail = shallow(
+			<SiteThumbnail mShotsUrl={ MSHOTS_URL } dimension={ dimension } />
+		);
+		const srcSet = siteThumbnail.find( IMG_SELECTOR ).prop( 'srcSet' );
+		const srcSet2xWith = ` ${ dimension.width * 2 }w`;
+		expect( srcSet.includes( srcSet2xWith ) ).toEqual( true );
+		const srcSet3xWith = ` ${ dimension.width * 3 }w`;
+		expect( srcSet.includes( srcSet3xWith ) ).toEqual( true );
+	} );
+
+	test( 'should add dimensionsSrcSet array porp to srcset string attribute', () => {
+		const alternativeDimensions = [
+			{
+				width: 777,
+				height: 777,
+			},
+			{
+				width: 888,
+				height: 888,
+			},
+		];
+		const siteThumbnail = shallow(
+			<SiteThumbnail mShotsUrl={ MSHOTS_URL } dimensionsSrcset={ alternativeDimensions } />
```

Comment:
Another approach would be to pass `width=""` and `height=""` to `<SiteThumbnail />`, such that `dimensionsSrcset` could then be an array of widths.

But, I don't have a strong opinion about it.

Reply:
@danielbachhuber
I like the suggestion to use `width` and `height` as props, instead of dimension. I've refactored `SiteThumbnail` to accept those props.
Since I'm refactoring a big part of the component, I've decided to make `alt` required, so we make sure our component follows the A11Y guidelines.
I updated the [component readme](https://github.com/Automattic/wp-calypso/pull/66988/files#diff-641c206e9a1ae875d814408209a9c975c663f482add5a5ab0d88fef3eb842cac) too.

As we discussed, we could deliver this PR when it's on production, we will be able to add the correct sizes to the grid thumbnail.

Could you make a final review?